src/file.js

```diff
@@ -1827,6 +1871,41 @@ class File extends common.ServiceObject {
       });
   }
 
+  /**
+   * Hold this file from its bucket's retention period configuration.
+   *
+   * By default, this will set an event-based hold. This is a way to retain
+   * objects until an event occurs, which is signified by the hold's release
+   * (i.e. this value is set to false (see @{link File#release})). After being
+   * released, this object will be subject to bucket-level retention (if any).
+   *
+   * Alternatively, you may set a temporary hold. This will follow the same
+   * behavior as an event-based hold, with the exception that the bucket's
+   * retention policy will not renew for this file from the time the hold is
+   * released.
+   *
+   * @param {object} [options] Configuration object.
+   * @param {boolean} [options.temporary=false] - Set a temporary hold.
+   * @param {SetFileMetadataCallback} [callback] Callback function.
+   * @returns {Promise<SetFileMetadataResponse>}
+   */
+  hold(options, callback) {
```

Comment:
This is subtle to me, is this considered idiomatic for Node.js?

Reply:
Would you mind elaborating?

Reply:
Sorry, I meant, given that each hold is different, could the surface API have separate methods?

Reply:
I like the idea of having these convenience methods, but it might be overdoing it as a user can update these holds using a setMetadata. What are your thoughts? I'm unsure honestly given the rest of the surface. Looking for input :)

cc: @mohammadnasre

Reply:
So disregard the mention of removing convenience methods. It makes sense here. 

Pending additional feedback from Mohammad.
I think, unifying methods is hard to understand, because at first glance I don't understand what kind of hold the method apply's. 

As a developer, I'd appreciate getting a hint in the name as to what the hold does given bucket lock surfaces two holds.

Reply:
I started fleshing this out into multiple convenient methods. I realized the methods were such simple calls to `setMetadata`, e.g.:

```js
file.setTemporaryHold = function () { file.setMetadata({ temporaryHold: true }) }
file.setEventBasedHold = function () { file.setMetadata({ eventBasedHold: true }) }

file.releaseTemporaryHold = function () { file.setMetadata({ temporaryHold: false }) }
file.releaseEventBasedHold = function () { file.setMetadata({ eventBasedHold: false }) }
```

Seeing that, I started to lean a bit towards skipping the convenience methods, as we're not really adding that much convenience. We could just show an example in the docs for `setMetadata`?

Reply:
I think it add obfuscation, and I appreciate your input. Move forward with documentation instead. This is something we shouldn't over complicate if it doesn't help at all. It's more of an alias that could also help introduce developers to manipulating other metadata.

Reply:
One additional note, system tests should verify how the client library behaves when holds are enabled and attempting to perform an overwrite and delete.

I have additional tests if you'd like to reference tests in Java [PR#3574](https://github.com/GoogleCloudPlatform/google-cloud-java/pull/3574).

Reply:
Friendly ping, @stephenplusplus.